dcrpool.go

```diff
@@ -226,7 +226,6 @@ func realMain() error {
 		}
 	}
 
-	db.Close()
 	mpLog.Info("dcrpool shut down.")
```

Comment:
Should probably defer this shutdown message as well, otherwise any logging which is included in the deferred tasks will  come **after** the shutdown message.

Reply:
Yeah, I was doing a separate PR that reworks the lifecycle of everything.  I can do it as part of this one though.

Reply:
Added a defer for overall shutdown and updated the existing shutdown message to make it clear it's referring to the hub.